### PR TITLE
FEATURE: Add visible to getCustomers and add tasks to getProjects

### PR DESCRIPTION
**Pull request by waldi**
did i make it right?

### CODE REVIEW

File: core/includes/classes/remote.class.php [hunk 1]
```diff
@@ -394,6 +394,38 @@ public function getProjects($apiKey)
 		}
 
         if (count($projects) > 0) {
+
+            $tempProjects = array();
+            foreach($projects as $project) {
+                $tasks = array();
+
+                // @FIXME
```
Comment (kevinpapst) [inline core/includes/classes/remote.class.php]: Why is there a FIXME?

**Comment by kevinpapst** [comment]
Can you 1) explain the FIXME 2) fix it or 3) remove it?

**Comment by waldi** [comment]
The FIXME is from the "getTasks"-Code. I didn't wrote it, neither don't i know why it's there.. should i still remove it?

**Comment by kevinpapst** [comment]
Okay, I checked the current remote code and do see where it comes from...
Actually, I really don't like copy & pasting (extract method would be the correct thing here). 

What I would like to see here is:
An extracted method to fetch the tasks by projectID, which then is used in both the getTasks() and getProjects() methods. 

Also, the getProjects($apiKey) signature should be changed to include a second and optional parameter $includeTasks =>  getProjects($apiKey, $includeTasks = false)

There are some programs out there using the remote API. They might break from that change, as it is not backward compatible. Do you want to include these changes?

**Comment by waldi** [comment]
ok. yeah i also thought about adding a extracted method. i'll do that.

yeah i'll do that change.

**Comment by kevinpapst** [comment]
:+1: thanks

**Comment by waldi** [comment]
ok i fixed it

**Pull request merged**